examples/simulation.py

```diff
@@ -18,13 +18,18 @@ def __init__(self, ipdb):
     # if called directly outside Simulation class, "ifc_base_name" should be
     # different from "name", the "ifc_base_name" and "name" are the same for
     # the first ifc created by namespace
-    def _ns_add_ifc(self, name, ns_ifc, ifc_base_name, in_ifc=None, out_ifc=None,
-                    ipaddr=None, macaddr=None, fn=None, cmd=None, action="ok",
-                    disable_ipv6=False):
+    def _ns_add_ifc(self, name, ns_ifc, ifc_base_name=None, in_ifc=None,
+                    out_ifc=None, ipaddr=None, macaddr=None, fn=None, cmd=None,
+                    action="ok", disable_ipv6=False):
         if name in self.ipdbs:
             ns_ipdb = self.ipdbs[name]
         else:
             ns_ipdb = IPDB(nl=NetNS(name))
+            if disable_ipv6:
+                cmd = ["sysctl", "-q", "-w", "net.ipv6.conf.default.disable_ipv6=1"]
+                nsp = NSPopen(ns_ipdb.nl.netns, cmd)
+                nsp.wait(); nsp.release()
+            ns_ipdb.interfaces.lo.up().commit()
```

Comment:
out_ifc is not in the namespace.

On Mon, Jul 13, 2015 at 9:57 PM, Brenden Blanco <notifications@github.com>
wrote:

> In examples/simulation.py
> <https://github.com/iovisor/bcc/pull/85#discussion_r34537240>:
>
> >          if name in self.ipdbs:
> >              ns_ipdb = self.ipdbs[name]
> >          else:
> >              ns_ipdb = IPDB(nl=NetNS(name))
> > +            if disable_ipv6:
> > +                cmd = ["sysctl", "-q", "-w", "net.ipv6.conf.default.disable_ipv6=1"]
> > +                nsp = NSPopen(ns_ipdb.nl.netns, cmd)
> > +                nsp.wait(); nsp.release()
> > +            ns_ipdb.interfaces.lo.up().commit()
>
> Don't all of the ifcs end up inside a netns?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/iovisor/bcc/pull/85/files#r34537240>.
>